pinot-common/src/test/java/org/apache/pinot/common/metrics/AbstractMetricsTest.java

```diff
@@ -268,4 +274,157 @@ public void testMeteredMetrics() {
     testMetrics.getMetricsRegistry().allMetrics().keySet().forEach(testMetrics.getMetricsRegistry()::removeMetric);
     Assert.assertTrue(testMetrics.getMetricsRegistry().allMetrics().isEmpty());
   }
+
+  @Test
+  public void testGlobalGaugeMetricsAsyncAddRemove() throws ExecutionException, InterruptedException {
+    ControllerMetrics controllerMetrics = buildTestMetrics();
+    int gaugeOperationsRuntimeMs = 100;
+
+    ExecutorService executorService = Executors.newFixedThreadPool(2);
+    long endTime = System.currentTimeMillis() + gaugeOperationsRuntimeMs;
+
+    Future<?> addFuture = executorService.submit(() -> {
+      while (System.currentTimeMillis() < endTime + gaugeOperationsRuntimeMs) {
+        controllerMetrics.addValueToGlobalGauge(ControllerGauge.VERSION, 1L);
```

Comment:
`addValueToGlobalGauge` is deprecated let's test `setValueOfGauge` instead.

Reply:
Since `addValueToGlobalGauge` has significant usage, I would keep that test and also add a separate async test for `setValueOfGauge`.